sdk/core/core-http/src/fetchHttpClient.ts

```diff
@@ -153,14 +154,20 @@ export abstract class FetchHttpClient implements HttpClient {
       const response: CommonResponse = await this.fetch(httpRequest.url, requestInit);
 
       const headers = parseHeaders(response.headers);
+
```

Comment:
@xirzec in core v2 this is in `core-https`. I don't think it deal with operation specs so we wouldn't be able to check the response type based on status code.  Any suggestions?

Reply:
Oh. So is the only way to detect the response shape based on what we think we know about the response from the operation spec? Seems like there should be a way to detect this without prior knowledge.

Reply:
It is the current way. Based on the types from the operation spec, we put the body into either `readableStreamBody`/`blobBody`, or `bodyAsText` property. I can think of some heuristics, for example,

- treat as text if content type header value is one of the several that we would parse, or is plain text. otherwise treat as stream.  For text cases we would store `response.text()` of the original response in `bodyAsText` so hopefully no matter how the body is transfer back we would have text at the end.

Anyway it feels like core v2 needs a different fix than porting this PR. I will log a separate issue to track it.

Reply:
logged #13235